File: iloczyn_macierzy.rb
wiersze1=0
wiersze2=0
kolumny1=0
kolumny2=0
i=0
j=0
liczba=0
puts  "Podaj ilosc wierszy pierwszej macierzy : "
wiersze1=gets.chomp.to_i
puts  "Podaj ilosc kolumn pierwszej macierzy : "
kolumny1=gets.chomp.to_i
puts  "Podaj ilosc wierszy drugiej macierzy : "
wiersze2=gets.chomp.to_i
puts  "Podaj ilosc kolumn drugiej macierzy : "
kolumny2=gets.chomp.to_i
macierz1 = Array.new(wiersze1) { Array.new(kolumny1, 0) }
macierz2 = Array.new(wiersze2) { Array.new(kolumny2, 0) }

if kolumny1==wiersze2
	puts "\n Wstawiaj liczby do pierwszej macierzy: "
		for i in 0..wiersze1-1
 			for j in 0..kolumny1-1
				macierz1[i][j]=gets.chomp.to_i
				j+=1
			end
		i+=1
	end
	puts "\n Wstawiaj liczby do drugiej macierzy: "
		for i in 0..wiersze2-1
 			for j in 0..kolumny2-1
				macierz2[i][j]=gets.chomp.to_i
				j+=1
			end
		i+=1
	end
	puts "\n\n Macierz pierwsza\n"
	for i in 0..wiersze1-1
		for j in 0..kolumny1-1
			print "#{macierz1[i][j]} "
			j+=1	
		end
		print "\n"
		i+=1
	end
	puts "\n\n Macierz druga\n"
	for i in 0..wiersze2-1
		for j in 0..kolumny2-1
			print "#{macierz2[i][j]} "
			j+=1	
		end
		print "\n"
		i+=1
	end


require_relative 'def_macierz'
mnozenie(wiersze1,kolumny1,wiersze2,kolumny2,macierz1,macierz2)
else
	puts "Mnozenie jest niewykonalne"
end    

File: Programy w C/iloczyn_macierzy.c
//Marcin Rzepniewski  -  Mnozenie Macierzy
#include<stdio.h>

int main(void)
{
	int wiersze1,wiersze2,kolumny1,kolumny2;
	printf("Podaj ilosc wierszy pierwszej macierzy");
	scanf("%d", &wiersze1);	
	printf("\nPodaj ilosc kolumn pierwszej macierzy");
	scanf("%d", &kolumny1);	
	printf("\nPodaj ilosc wierszy drugiej macierzy");
	scanf("%d", &wiersze2);	
	printf("\nPodaj ilosc kolumn drugiej macierzy");
	scanf("%d", &kolumny2);
	int macierz1[wiersze1][kolumny1];
	int macierz2[wiersze2][kolumny2];
	int macierzwynikowa[wiersze1][kolumny2];
	if(kolumny1==wiersze2)
	{	
		printf("\nWstawiaj liczby do pierwszej macierzy");
		int i,j;
		for(i=0;i<wiersze1;i++)
		{
			for(j=0;j<kolumny1;j++)
			{
				scanf("%d", &macierz1[i][j]);
			}
		}

		printf("\nWstawiaj liczby do drugiej macierzy");
        	for(i=0;i<wiersze2;i++)
        	{
        	        for(j=0;j<kolumny2;j++)
        	        {
        	                scanf("%d", &macierz2[i][j]);
        	        }
        	}
       	
		printf("\n\n Macierz pierwsza\n");
        	for(i=0;i<wiersze1;i++)
		{
			for(j=0;j<kolumny1;j++)
			{
				printf("%d\t", macierz1[i][j]);
			}
			printf("\n");
		}
		printf("\n\n Macierz druga\n");
        for(i=0;i<wiersze2;i++)
        {
                for(j=0;j<kolumny2;j++)
		{
			printf("%d\t", macierz2[i][j]);
		}
		printf("\n");
	}
	int k;
	int wynik=0;
	// mnozenie macierzy i zapis do nowej tablicy 
	for(i=0;i<wiersze1;i++){	
	for(j=0;j<kolumny2;j++)
		{
			macierzwynikowa[i][j]=0;
			for(k=0;k<wiersze2;k++)
			{
		//		printf("wynik = %d + %d*%d\n", macierzwynikowa[i][j],macierz1[i][k],macierz2[k][j]);

				macierzwynikowa[i][j]+=macierz1[i][k]*macierz2[k][j];
//				printf("wynik = %d + %d*%d\n", macierzwynikowa[i][j]+macierz1[j][k],macierz2[k][j]);
			}	
		printf("\n\n NEXT\n\n");	
		}
	}	
	
printf("\n\n\n MAcierz wynikowa \n\n");
  for(i=0;i<wiersze1;i++)
	{
		for(j=0;j<kolumny2;j++)
		{
			printf("%d\t", macierzwynikowa[i][j]);
		}
		printf("\n");
	}

}
	else
	{
		printf("\nPodane macierze nie da sie mnozyc\n");
	}

	return 0;
}
